app/controllers/rooms_controller.rb

```diff
@@ -1,17 +1,19 @@
 class RoomsController < ApplicationController
+  respond_to :json, :html
+
   def new
   end
 
   def create
     room = Room.new(room_params)
-    room.choices.build(title: params[:room][:choice])
+    params[:room][:choices].each { |choice| room.choices.build(title: choice) }
     if room.save
       room.update!(user_id: current_user.id)
-      redirect_to room_path(room)
-      flash[:success] = "Your room has been created!"
+      respond_with room
+      # redirect_to room_path(room)
+      # flash[:success] = "Your room has been created!"
     else
-      flash[:error] = "An error prevented your room from being saved."
-      redirect_to root_path
+      head :bad_request
```

Comment:
Surely, I have used this with ajax quite a bit. Head builds responses that have an empty body, just a header.  In the case of :bad_request, it will send a status code of 400 when this else condition is hit. You can see this in the browser console if you make an invalid room. Its been handy to use when you do not want to redirect or render.   https://github.com/rack/rack/blob/master/lib/rack/utils.rb#L402   -   you can scroll down and look at all the handy http status codes with their corresponding strings